syncd/syncd_flex_counter.cpp

```diff
@@ -188,6 +188,8 @@ void FlexCounter::setPortCounterList(
     auto portCounterIds = std::make_shared<PortCounterIds>(portId, supportedIds);
     fc.m_portCounterIdsMap.emplace(portVid, portCounterIds);
 
+    fc.m_collectCountersHandlers.emplace(PORT_COUNTER_ID_LIST, &FlexCounter::collectPortCounters);
```

Comment:
if you look from this line to below it's look like copy paste a lot of statements
maybe it could be optimized/merged to some array or function

Reply:
Not quite get the way for generalization. Wrap the emplace/erase inside a member function is what I can think of. Please provide more details about your idea.